share/eds/store_test.go

```diff
@@ -122,6 +122,37 @@ func TestEDSStore(t *testing.T) {
 		assert.ErrorContains(t, err, "no such file or directory")
 	})
 
+	t.Run("Remove after OpShardFail", func(t *testing.T) {
+		eds, dah := randomEDS(t)
+
+		err = edsStore.Put(ctx, dah.Hash(), eds)
+		require.NoError(t, err)
+
+		// assert that shard now exists
+		ok, err := edsStore.Has(ctx, dah.Hash())
+		assert.NoError(t, err)
+		assert.True(t, ok)
+
+		// assert that file now exists
+		path := edsStore.basepath + blocksPath + dah.String()
+		_, err = os.Stat(path)
+		assert.NoError(t, err)
+
+		err = os.Remove(path)
+		assert.NoError(t, err)
+
+		_, err = edsStore.GetCAR(ctx, dah.Hash())
+		assert.Error(t, err)
+
+		// give some time for the store to remove
+		time.Sleep(300 * time.Millisecond)
```

Comment:
it might be nice to wrap this in a for {} and actually check that the path you deleted in os.Remove above is actually clear before proceeding to the next check, vs relying on the time.Sleep which could be a intermittent flakey test later. Say a runner is slow one time, or for some reason THIS time it look 301ms, or something else.

maybe something like 

```go

_, err := os.Remove(path)

maxAttemps := 5
attempts := 0
wait := 300 * time.Millisecond

for {
  _, err := os.Stat(path)
  if err != nil {
    if !os.IsNotFound(err) {
      break
    }   
    t.Error(err)
  }
  
  attempts++
  if retries >= maxAttempts {
     break
  } 

  time.Sleep(wait)
}

```

we could potentially max the 300ms much shorter and proceed through the test quicker than the 300ms in most cases too

Reply:
Yeah, this was a bit lazy from me. Will change to a `select` with a ticker that will just repeat the check until the test context deadlines